src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/RowTests.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Linq;
+using Microsoft.Spark.Sql;
+using Xunit;
+using static Microsoft.Spark.Sql.Functions;
+
+namespace Microsoft.Spark.E2ETest.IpcTests
+{
+    [Collection("Spark E2E Tests")]
+    public class RowTests
+    {
+        private readonly SparkSession _spark;
+        private readonly DataFrame _df;
+
+        public RowTests(SparkFixture fixture)
+        {
+            _spark = fixture.Spark;
+            _df = _spark
+                .Read()
+                .Schema("age INT, name STRING")
+                .Json($"{TestEnvironment.ResourceDirectory}people.json");
+        }
+
+        [Fact]
+        public void TestCollect()
+        {
+            Row[] rows = _df.Collect().ToArray();
+            Assert.Equal(3, rows.Length);
+
+            Row row1 = rows[0];
+            Assert.Equal("Michael", row1.GetAs<string>("name"));
+            Assert.Null(row1.Get("age"));
+
+            Row row2 = rows[1];
+            Assert.Equal("Andy", row2.GetAs<string>("name"));
+            Assert.Equal(30, row2.GetAs<int>("age"));
+
+            Row row3 = rows[2];
+            Assert.Equal("Justin", row3.GetAs<string>("name"));
+            Assert.Equal(19, row3.GetAs<int>("age"));
+        }
+
+        [Fact]
+        public void TestWithColumn()
```

Comment:
You are testing how Row is being passed? o.w. this should belong to `DataFrameTests`. Any particular reason it's move to its own test suite?

Reply:
Moved everything to `DataFrameTests`